datafusion/substrait/src/logical_plan/producer.rs

```diff
@@ -368,14 +368,45 @@ pub fn to_substrait_rel(
                 .iter()
                 .map(|e| substrait_sort_field(state, e, sort.input.schema(), extensions))
                 .collect::<Result<Vec<_>>>()?;
-            Ok(Box::new(Rel {
+
+            let sort_rel = Box::new(Rel {
                 rel_type: Some(RelType::Sort(Box::new(SortRel {
                     common: None,
                     input: Some(input),
                     sorts: sort_fields,
                     advanced_extension: None,
                 }))),
-            }))
+            });
+
+            match sort.fetch {
+                Some(_) => {
+                    let empty_schema = Arc::new(DFSchema::empty());
+                    let count_mode = sort
+                        .fetch
+                        .map(|amount| {
```

Comment:
given you have the `match sort.fetch { Some(_) =>` above, you could store that "some" and then use it here
```
match sort.fetch {
   Some(amount) => { 
      ...
      let count_mode = to_substrait_rex(.., (amount as i64), ...)
```

Reply:
also I think rather than using to_substrait_rex, you could create the substrait expr directly, that might be nicer here

Reply:
ty for the suggestions!  Heading to dinner now and I can address these tomorrow morning eastern time.

Reply:
Incorporated both suggestions.  Thank you @Blizzara